classes/class-db-driver-wpdb.php

```diff
@@ -81,7 +81,9 @@ public function insert_record( $data ) {
 		// Insert record meta.
 		foreach ( (array) $meta as $key => $vals ) {
 			foreach ( (array) $vals as $val ) {
-				$this->insert_meta( $record_id, $key, $val );
+				if ( $val ) {
```

Comment:
Should we allow `0` (int) and `'0'` (string) values to be stored? Maybe check for `null` and `''` (empty string)?

Reply:
I found this note right above this link anchor https://developer.wordpress.org/reference/classes/wpdb/#examples-6

> Please note: The value portion of the data parameter’s column=>value pairs must be scalar. If you pass an array (or object) as a value to be inserted you will generate a warning.

Hence, I think getting `is_scalar()` involved here is the right move to make.

Only remaining thing I'm wondering about now: does boolean `false` also get converted to empty string, or TINYINT 0?

> If omitted, all values in $data will be treated as strings unless otherwise specified in wpdb::$field_types.

Should our sanity check be `if ( ... && (string) $val !== '' )`?